src/player_hardcoded_effects.cpp

```diff
@@ -1110,6 +1111,45 @@ void player::hardcoded_effects( effect &it )
                     }
                 }
             }
+            if( ( ( has_trait( trait_id( "SCHIZOPHRENIC" ) ) || has_artifact_with( AEP_SCHIZO ) ) && one_in( 7200 ) && is_player() ) ) {
+                if( one_in( 2 ) ) {
+                    sound_hallu();
+                } else {
+                    int x = posx();
+                    int y = posy();
+                    int z = posz();
+
+                    tripoint mp = tripoint( x, y, z );
```

Comment:
Use `mp = pos();` instead of getting the x, y, z components separately and than putting them back into a `tripoint`.